parts/kubernetesmastercustomscript.sh

```diff
@@ -22,6 +22,19 @@
 # APISERVER_PRIVATE_KEY CA_CERTIFICATE CA_PRIVATE_KEY MASTER_FQDN KUBECONFIG_CERTIFICATE
 # KUBECONFIG_KEY ADMINUSER
 
+# Handling for different OSes
+shopt -s nocasematch
+OS=$(cat /etc/*release | grep ^NAME | tr -d 'NAME="')
```

Comment:
Would it be possible to do something like this instead?

OS="UBUNTU"
shopt -s noscasematch
if [[ `$(cat /etc/*release| grep ^NAME | tr -d 'NAME="'` ==*"COREOS"* ]]; then
    OS = "COREOS"
 fi
shopt +s noscasematch     // turn off nocasematch

continue on including lines 33-36

This would ensure there are no side effects from doing shopt -s nocasematch

Reply:
I've added code that doesn't use the case match setting and worrying about side effects make sense.

```bash
OS=$(cat /etc/*-release | grep ^ID= | tr -d 'ID="' | awk '{print toupper($0)}')
UBUNTU_OS_NAME="UBUNTU"
RHEL_OS_NAME="RHEL"
COREOS_OS_NAME="COREOS"
```